sensors/sqs_service_sensor.py

```diff
@@ -0,0 +1,155 @@
+"""
+This is generic SQS Sensor using boto3 api to fetch messages from sqs queue.
+After receiving a message it's content is passed as payload to a trigger 'aws.sqs_new_message'
+This sensor can be configured either by using config.yaml within a pack or by creating
+following values in datastore:
+    - aws.input_queues (list queues as comma separated string: first_queue,second_queue)
+    - aws.aws_access_key_id
+    - aws.aws_secret_access_key
+    - aws.region
+    - aws.max_number_of_messages (must be between 1 - 10)
+For configuration in config.yaml with config like this
+    setup:
+      aws_access_key_id:
+      aws_access_key_id:
+      region:
+    sqs_sensor:
+      input_queues:
+        - first_queue
+        - second_queue
+    sqs_other:
+        max_number_of_messages: 1
+If any value exist in datastore it will be taken instead of any value in config.yaml
+"""
+
+import six
+import json
+from boto3.session import Session
+from botocore.exceptions import ClientError
+from botocore.exceptions import NoRegionError
+from botocore.exceptions import NoCredentialsError
+from botocore.exceptions import EndpointConnectionError
+
+from st2reactor.sensor.base import Sensor
+
+
+class AWSSQSServiceSensor(Sensor):
+    def __init__(self, sensor_service, config=None):
+        super(AWSSQSServiceSensor, self).__init__(sensor_service=sensor_service, config=config)
+
+    def setup(self):
+        self._logger = self._sensor_service.get_logger(name=self.__class__.__name__)
+
+        self.session = None
+        self.sqs_res = None
+
+    def run(self):
+        # setting SQS ServiceResource object from the parameter of datastore or configuration file
+        self._may_setup_sqs()
+
+        while True:
+            for queue in self.input_queues:
+                msgs = self._receive_messages(queue=self._get_queue_by_name(queue),
+                                            num_messages=self.max_number_of_messages)
+                for msg in msgs:
+                    if msg:
+                        payload = {"queue": queue, "body": json.loads(msg.body)}
+                        self._sensor_service.dispatch(trigger="aws.sqs_new_message",
+                                                      payload=payload)
+                        msg.delete()
```

Comment:
Can this method throw?

If so, it probably wouldn't be a bad idea to wrap it in try / catch to avoid a scenario where the same message would always throw for some reason which would prevent sensor from continuing the processing since it would always crash on exception...

Reply:
If this throws an exception, it's not because SQS couldn't delete the message due to some unrecoverable error. It will be because of a misconfiguration on the client side or something like that. I don't think it is necessary to try/catch this.